### ISSUE TITLE
Format of input data

### ISSUE THREAD

**Issue by ooobsidian** [BUTSpeechFIT/EEND#5]
Excellent code! I see that the input data needs to be filled in with "\<Kaldi data directory for train set\>", but I don't know how to prepare the data for the kaldi type.

I look forward to hearing from you

**Comment by fnlandini** [comment]
Hi, thank you for the feedback
What I meant with that is that you need to pass a directory that contains files reco2dur, rttm, segments, spk2utt, utt2spk, wav.scp following the usual format as done in the Kaldi toolkit.
For telephone data, you can look at this [script](https://github.com/hitachi-speech/EEND/blob/master/egs/callhome/v1/run_prepare_shared_eda.sh). However, we have prepared recently other examples on public data such as LibriSpeech and you can find the script [here](https://github.com/BUTSpeechFIT/EEND_dataprep/blob/main/v2/LibriSpeech/prepareKaldidata_LibriSpeech.sh) which I believe is easier to follow because there is a single corpus.
Let me know if this helps.

**Comment by ooobsidian** [comment]
Thank you for your reply. By the way, can EEND directly use the original speech wav and RTTM for supervised training, or what needs to be modified for adaptation, because I noticed that this warehouse is implemented based on Pytorch.

**Comment by fnlandini** [comment]
Hi, I am not sure if I understand so correct me if I am wrong if I understood you wrong.
This model is fully trained supervisedly so there is no pre-training that requires later adaptation. However, the model is trained on synthetic data first and then there is a fine-tuning step where one re-trains the model using (a small learning rate and) a small development set of real data. However, this is not related to the model being implemented in pytorch. This was the same in the original implementation in chainer.

**Comment by ooobsidian** [comment]
Thank you for your reply. I mean, can we train without converting the original data to Kaldi format data, or what changes should be made to the code?

**Comment by fnlandini** [comment]
You do not need to convert any data at all. By "Kaldi data directory" I mean that you need to create the files I mentioned before (reco2dur, rttm, segments, spk2utt, utt2spk, wav.scp) which basically list the files and segments of speech that one needs as training data. But the data are not changed. 
I will have to point you again to the scripts I mentioned before. You can study them or run them to get better understanding about how you need to read your data to produce the needed files.

**Comment by ooobsidian** [comment]
Thank you very much for your patience!

**Comment by fnlandini** [comment]
Closing. Feel free to reopen if you see fit.
